apps/utils/locust_tests/v2/partially_adjudicated_post.py

```diff
@@ -0,0 +1,124 @@
+import random
+from typing import Collection
+
+from locust import events, tag, task
+from locust.env import Environment
+
+from common import data, db
+from common.bfd_user_base import BFDUserBase
+from common.locust_utils import is_distributed, is_locust_master
+from common.user_init_aware_load_shape import UserInitAwareLoadShape
+
+master_pac_mbis: Collection[str] = []
+
+
+@events.test_start.add_listener
+def _(environment: Environment, **kwargs):
+    if (
+        is_distributed(environment)
+        and is_locust_master(environment)
+        or not environment.parsed_options
+    ):
+        # Don't bother loading data for the master runner, it doesn't run a test
+        return
+
+    # See https://docs.locust.io/en/stable/extending-locust.html#test-data-management
+    # for Locust's documentation on the test data management pattern used here
+    global master_pac_mbis
+    master_pac_mbis = data.load_from_parsed_opts(
+        environment.parsed_options,
+        db.get_pac_mbis,
+        use_table_sample=False,
+        data_type_name="pac_mbis",
+    )
+
+
+class TestLoadShape(UserInitAwareLoadShape):
+    pass
+
+
+class PACUser(BFDUserBase):
+    """
+    Tests for Partially Adjudicated Claims endpoints to test their performance
+
+    The MBI list is randomly shuffled to get better sampling in sequential testing.
+    """
+
+    SERVICE_DATE = {"service-date": "gt2020-01-05"}
+    LAST_UPDATED = {"_lastUpdated": "gt2020-05-05"}
+    SERVICE_DATE_LAST_UPDATED = dict(SERVICE_DATE, **LAST_UPDATED)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.mbis = list(master_pac_mbis)
+
+    def __post(self, resource, name, body=None):
+        params = {}
+        body = {} if body is None else body
+        body["mbi"] = random.choice(self.mbis)
+
+        self.run_task_by_parameters(
+            base_path=f"/v2/fhir/{resource}", params=params, body=body, name=name
+        )
+
+    @tag("claim")
+    @task
+    def get_claim(self):
+        """Get single Claim"""
+        self.__post("Claim/_search", "claim")
+
+    @tag("claim", "service-date")
+    @task
+    def get_claim_with_service_date(self):
+        """Get single Claim with service date"""
+        self.__post("Claim/_search", "claimServiceDate", self.SERVICE_DATE)
+
+    @tag("claim", "last-updated")
+    @task
+    def get_claim_with_last_updated(self):
+        """Get single Claim with last updated"""
+        self.__post("Claim/_search", "claimLastUpdated", self.LAST_UPDATED)
+
+    @tag("claim", "service-date", "last-updated")
+    @task
+    def get_claim_with_service_date_and_last_updated(self):
+        """Get single Claim with last updated and service date"""
+        self.__post("Claim/_search", "claimServiceDateLastUpdated", self.SERVICE_DATE_LAST_UPDATED)
+
+    @tag("claim", "exclude-samsa")
+    @task
+    def get_claim_with_exclude_samsa(self):
+        self.__post("Claim/_search", "excludeSAMSA", True)
```

Comment:
Looks like a typo (or propagated typo) here, too.

Reply:
It looks like that's actually an issue with the existing code lol. So we haven't been testing with `excludeSAMHSA` set properly at all... fun.

Reply:
Good catch. yes, this two excludeSAMHSA tests don't pass for both GET and POST. It was easily missed because it doesn't tell you a test failed either, You have to look at the log above and you will see errors related to it.
I do see issues with hitting both `Claim` and `ClaimResponse` endpoints even after correcting the `excludeSAMHSA` typo. I will start a thread

Reply:
This test files were old and didn't need a post version since they aren't  being used anymore, The new test files are deleted now. Thanks to Austin!